packages/connect/src/api/getPublicKey.ts

```diff
@@ -113,6 +113,11 @@ export default class GetPublicKey extends AbstractMethod<'getPublicKey', Params[
     }
 
     async noBackupConfirmation() {
+        // Do not show warning modal when the public key is not requested by user.
+        // E.g. it is accessed during creation of a coinjoin account.
+        if (!this.params[0].show_display) {
```

Comment:
maybe it would be better to check whole bundle? like:
if (!this.params.some(p => p.show_display))

Reply:
once it's done we can remove this workaround https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/middlewares/wallet/discoveryMiddleware.ts#L33-L47

Reply:
Checking the whole bundle in b4acfb0a4969c96d77fd32381543f07eb7ba0381

Regarding the workaround, I did not find a reason to have `noBackupConfirmation` on `TrezorConnect.getAccountInfo`, perhaps it should be removed to get rid of the workaround?

https://github.com/trezor/trezor-suite/blob/db702eaeb7ee0a918db2c1c1d131f3f23624512f/packages/connect/src/api/getAccountInfo.ts#L164-L180

Reply:
ah, right, i forgot that this is requested by getAccountInfo not by getPublicKey

i know one good reason why to have it there. this method is also used by 3rd party popup implementations and we want to show this warning in that case. But maybe we can disable it in trusted mode, what do you think @mroz22 ?

Reply:
i'd rather add a new parameter "suppressNoBackupConfirmation: bool"

Reply:
@mroz22 Ok, I will add the param to `getPublicKey`, `getAccountInfo` and `getAddress` and set it to `true`in all calls from Suite except when user displays an address or public key.

It should also resolve #5280.

Reply:
Done, can you take a look, @mroz22?